### PR TITLE
API: allow list-like to DataFrame rename

### LINKED ISSUE #14829: API: allow DataFrame.rename to take a list-like of colums
This is closely related to #12392, but I think separate issue.  Proposal would be to be able to pass a list-like to `DataFrame.rename` to make method-chaining easier.  I think it would also be consistent with #11980 (Series rename)

```python
df = pd.DataFrame({'a': [1,2], 'b': [3, 4]})

# make this
df = df.rename(columns=['j', 'k'])

# equivalent to 
df.columns = ['j', 'k']
```

### PR DESCRIPTION
**Pull request by chris-b1**
- [x] closes #14829
 - [ ] tests added / passed
 - [x] passes ``git diff upstream/master | flake8 --diff``
 - [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/test_generic.py [hunk 1]
```diff
@@ -1504,6 +1504,26 @@ def test_to_xarray(self):
...
+        result = df.rename(columns=['J', 'K'])
+        expected.columns = ['J', 'K']
+        assert_frame_equal(result, expected)
+
+        result = df.rename(columns=['J', 'K'], index=['a', 'b'])
+        expected.index = ['a', 'b']
+        assert_frame_equal(result, expected)
+
+        result = df.copy()
+        result.rename(columns=['J', 'K'], index=['a', 'b'], inplace=True)
```
Comment (jreback) [inline pandas/tests/test_generic.py]: test with ndarray and with an Index
a scalar should error yes?

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
this is also now an alternative to the ``.index =`` API (which is fine I think). So can you put some references in the doc-string and update the docs (where .set_index is).

**Comment by chris-b1** [comment]
I added a couple doc references, but I might be missing something - wasn't sure if there is a canonical spot for the `.index/.columns=` api?

**Comment by jreback** [comment]
http://pandas.pydata.org/pandas-docs/stable/indexing.html#set-reset-index

ad-hoc indexing. This would be expanded in any event (and *maybe* moved to http://pandas.pydata.org/pandas-docs/stable/basics.html#renaming-mapping-labels) as these are the same idea.

**Comment by chris-b1** [comment]
Closing in favor of #15104

**Pull request closed without merging**